include/FieldDefinitions.h

```diff
@@ -29,6 +29,7 @@ using FieldDefGenericInt = FieldDefinition<GenericIntFieldType>;
 using FieldDefTpetraId = FieldDefinition<TpetIDFieldType>;
 using FieldDefLocalId = FieldDefinition<LocalIdFieldType>;
 using FieldDefGlobalId = FieldDefinition<GlobalIdFieldType>;
+using FieldDefHypreId = FieldDefinition<HypreIDFieldType>;
```

Comment:
Somehow I hadn't noticed FieldDefinition previously. It seems to just add rank and num-states to field, but field already holds those attributes. i.e., each field already knows its rank and num-states.

Reply:
Yes, once the field has been created it knows this information. The registry exists to define all the information about the field needed to create it. The end intent is that a field can be registered and populated to parts from a single call that only requires the field's name as a string. That is when the field actually gets registered through the `FieldManager::register_field` call.